cmd/dm-syncer/config.go

```diff
@@ -0,0 +1,383 @@
+	fs.IntVar(&cfg.Batch, "b", 100, "batch commit count")
+	fs.StringVar(&cfg.StatusAddr, "status-addr", ":8271", "status addr")
+	fs.StringVar(&cfg.Meta, "meta", "syncer.meta", "syncer meta info")
+	//fs.StringVar(&cfg.PersistentTableDir, "persistent-dir", "", "syncer history table structures persistent dir; set to non-empty string will choosing history table structure according to column length when constructing DML")
+	fs.StringVar(&cfg.LogLevel, "L", "info", "log level: debug, info, warn, error, fatal")
+	fs.StringVar(&cfg.LogFile, "log-file", "", "log file path")
+	fs.StringVar(&cfg.LogRotate, "log-rotate", "day", "log file rotate type, hour/day")
+	fs.BoolVar(&cfg.EnableGTID, "enable-gtid", false, "enable gtid mode")
+	fs.BoolVar(&cfg.SafeMode, "safe-mode", false, "enable safe mode to make syncer reentrant")
+	fs.IntVar(&cfg.MaxRetry, "max-retry", 100, "maxinum retry when network interruption")
+	fs.BoolVar(&cfg.EnableANSIQuotes, "enable-ansi-quotes", false, "enable ANSI_QUOTES sql_mode")
+	fs.StringVar(&cfg.TimezoneStr, "timezone", "", "target database timezone location string")
+	fs.BoolVar(&OldConfigFormat, "old-config-format", false, "read old config format")
+
+	fs.Parse(args)
+
+	if cfg.ConfigFile != "" {
+		_, err := toml.DecodeFile(cfg.ConfigFile, cfg)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+
+	fs.Parse(args)
+
+	return cfg.convertToNewFormat()
+}
+
+func (c *commonConfig) parse(args []string) (*config.SubTaskConfig, error) {
+	err := c.FlagSet.Parse(args)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if c.printVersion {
+		fmt.Println(utils.GetRawInfo())
+		return nil, flag.ErrHelp
+	}
+
+	if c.OldConfigFormat {
+		return c.newConfigFromOldConfig(args)
+	}
+
+	return c.newSubTaskConfig(args)
+}
+
+func (c *commonConfig) newSubTaskConfig(args []string) (*config.SubTaskConfig, error) {
+
+	cfg := &config.SubTaskConfig{}
+	cfg.SetFlagSet(flag.NewFlagSet("common-syncer", flag.ContinueOnError))
+	fs := cfg.GetFlagSet()
+
+	var oldConfigFormat bool
+	var printVersion bool
+	var serverID uint
+
+	fs.BoolVar(&printVersion, "V", false, "prints version and exit")
+	fs.StringVar(&cfg.ConfigFile, "config", "", "path to config file")
+	fs.UintVar(&serverID, "server-id", 101, "MySQL slave server ID")
+	fs.StringVar(&cfg.Flavor, "flavor", mysql.MySQLFlavor, "use flavor for different MySQL source versions; support \"mysql\", \"mariadb\" now; if you replicate from mariadb, please set it to \"mariadb\"")
+	fs.IntVar(&cfg.WorkerCount, "c", 16, "parallel worker count")
+	fs.IntVar(&cfg.Batch, "b", 100, "batch commit count")
+	fs.StringVar(&cfg.StatusAddr, "status-addr", ":8271", "status addr")
+	//fs.StringVar(&cfg.PersistentTableDir, "persistent-dir", "", "syncer history table structures persistent dir; set to non-empty string will choosing history table structure according to column length when constructing DML")
+	fs.StringVar(&cfg.MetaFile, "meta", "syncer.meta", "syncer meta info")
```

Comment:
We do not use `meta` in DM anymore, how about removing it?

Reply:
ok, remove it in order to avoid confusing